src/components/Results.js

```diff
@@ -171,24 +268,145 @@ class Results extends Component {
+
+              return {
+                key: idResult,
+                name: nameResult,
+                algorithmName: algorithmNameResult,
+                result: (
+                  <>
+                    <Grid container alignItems="center" spacing={1}>
+                      <Grid item>
+                        <Typography variant="subtitle1" key="name">
+                          {nameResult}
+                        </Typography>
+                      </Grid>
+                      <SchemaTags schemaIds={schemaResultIds} />
+                    </Grid>
+                    <Typography variant="caption" key="description">
+                      {descriptionResult}
+                    </Typography>
+                  </>
+                ),
+                size: sizeStringResult,
+                sizeNumeric: sizeResult,
+                createdAt: createdAtStringResult,
+                lastModified: lastModifiedStringResult,
+                quickActions: [
+                  <ViewDatasetButton
+                    tooltip={t('View result')}
+                    key="view"
+                    dataset={res}
+                  />,
+                  <ExportButton
+                    id={id}
+                    name={`${name}.json`}
+                    flagType={FLAG_EXPORTING_RESULT}
+                    channel={EXPORT_RESULT_CHANNEL}
+                    tooltipText={t('Export result')}
+                  />,
+                  <Tooltip title={t('Remove result')} key="delete">
+                    <IconButton
+                      aria-label="delete"
+                      onClick={() => this.handleDelete(res)}
+                    >
+                      <DeleteIcon />
+                    </IconButton>
+                  </Tooltip>,
+                  <Tooltip title={t('Edit result')} key="edit">
+                    <IconButton
+                      disabled
+                      aria-label="edit"
+                      onClick={() => this.handleEdit(res)}
+                    >
+                      <EditIcon />
+                    </IconButton>
+                  </Tooltip>,
+                  <Tooltip title={t('Publish result')} key="publish">
+                    <IconButton
+                      disabled
+                      aria-label="publish"
+                      onClick={() => this.handlePublish(res)}
+                    >
+                      <PublishIcon />
+                    </IconButton>
+                  </Tooltip>,
+                ],
+              };
+            })
+          : undefined;
 
       return {
         key: id,
         name,
         algorithmName,
+        collapse:
+          result.length > 1 ? (
+            <IconButton
+              aria-label="expand row"
+              size="small"
+              id={buildExecutionCollapsePipelineButtonId(resultIdx)}
+              onClick={() => {
+                const switchCollapsePipeline = [...collapsePipeline];
```

Comment:
After some inspection of your code, option 2 seems to fit well (for both executions and results), we can go through it together.

Reply:
I don't really understand the second option. Since the results can be deleted,  `collapsePipeline`'s length changes, and because of that, we need to remove the useless boolean states accordingly.